File: FundTrack.WebUI/Controllers/AuthorizationType.cs
using System.Collections.Generic;
using FundTrack.Infrastructure.ViewModel;

namespace FundTrack.WebUI.Controllers
{
    internal class AuthorizeUserModel
    {
        public string login { get; set; }
        public int id { get; set; }
        public string firstName { get; set; }
        public string lastName { get; set; }
        public string email { get; set; }
        public string address { get; set; }
        public string photoUrl { get; set; }
        public string role { get; set; }
    }

    internal class AuthorizationType
    {
        public AuthorizeUserModel userModel { get; set; }
        public string access_token { get; set; }
        public string errorMessage { get; internal set; }
        public List<ValidationViewModel> validationSummary { get; internal set; }
    }
}

File: FundTrack.WebUI/ClientApp/app/view-models/concrete/authorization.type.ts
/**
 * model which used when return authorize user
 */
import { ValidationViewModel } from "./validation-view.model";

export class AuthorizeUserModel {
    public login: string;
    public id: number;
    public firstName: string;
    public lastName: string;
    public email: string;
    public address: string;
    public photoUrl: string;
    public role: string;
}
export class AuthorizationType {
    constructor(
        public userModel: AuthorizeUserModel,
        public access_token: string,
        public errorMessage: string,
        public validationSummary: ValidationViewModel[]
    )
    { }
}
